### PR TITLE
Implement Lasso Utils

### PR DESCRIPTION
**Pull request by Jiarui-Cao**
### Description
Utils for Lasso container:
- util that converts `asset` to model filename
- util that checks whether a directory has model for given assets

### Testing
- Unit testing for first util (convert string filename)
- Integration testing for util that searches directory

### Impact
Prereq for lasso container

### CODE REVIEW

File: .DS_Store
Comment (echavemann) [inline .DS_Store:1]: no

File: src/pysrc/test/integration/test_lasso_utils.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+from pysrc.util.types import Asset
+from pysrc.util.lasso_utils import check_lasso_models
+from pysrc.test.helpers import get_resources_path
+from pathlib import Path
+
+
+def test_check_lasso_models() -> None:
```
Comment (echavemann) [inline src/pysrc/test/integration/test_lasso_utils.py:7]: id seperate the positive and negative assertions

File: src/pysrc/test/integration/test_lasso_utils.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    resource_path = get_resources_path(str(Path(__file__).resolve()))
```
Comment (echavemann) [inline src/pysrc/test/integration/test_lasso_utils.py:8]: you should be able to call get_resources_path(__file__)

Comment (echavemann) [reply src/pysrc/test/integration/test_lasso_utils.py:8]: `__file__`

File: src/pysrc/util/lasso_utils.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+from pysrc.util.types import Asset
+from pathlib import Path
+
+def get_lasso_model_filename(asset: Asset) -> str:
+    return(str(asset.name) + "_lasso.pkl")
+
+def check_lasso_models(directory_path: Path, assets: list[Asset]) -> bool:
```
Comment (echavemann) [inline src/pysrc/util/lasso_utils.py:7]: better name

File: src/pysrc/util/lasso_utils.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+from pysrc.util.types import Asset
+from pathlib import Path
+
+def get_lasso_model_filename(asset: Asset) -> str:
```
Comment (echavemann) [inline src/pysrc/util/lasso_utils.py:4]: lets base on Aidans branch and use his new enum function

File: .DS_Store
Comment (echavemann) [reply .DS_Store:1]: add to gitignore while youre at it

File: src/pysrc/util/lasso_utils.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
+from pysrc.util.types import Asset
+from pysrc.util.enum_conversions import enum_to_string
+from pathlib import Path
+
+def asset_to_model_filename(asset: Asset) -> str:
+    return(enum_to_string(asset) + "_lasso.pkl")
```
Comment (echavemann) [inline src/pysrc/util/lasso_utils.py:6]: more idiomatic to use fstring here

**Review by echavemann — APPROVED** [review]
nice

File: src/pysrc/util/lasso_utils.py (hunk 5 above)
Comment (Jiarui-Cao) [reply src/pysrc/util/lasso_utils.py]: fixed

File: src/pysrc/util/lasso_utils.py (hunk 4 above)
Comment (Jiarui-Cao) [reply src/pysrc/util/lasso_utils.py]: rebased

File: src/pysrc/util/lasso_utils.py (hunk 3 above)
Comment (Jiarui-Cao) [reply src/pysrc/util/lasso_utils.py]: renamed

File: src/pysrc/test/integration/test_lasso_utils.py (hunk 2 above)
Comment (Jiarui-Cao) [reply src/pysrc/test/integration/test_lasso_utils.py]: changed

File: src/pysrc/test/integration/test_lasso_utils.py (hunk 1 above)
Comment (Jiarui-Cao) [reply src/pysrc/test/integration/test_lasso_utils.py]: created 2 functions

File: .DS_Store
Comment (Jiarui-Cao) [reply .DS_Store:1]: updated gitignore

File: src/pysrc/util/lasso_utils.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
+from pysrc.util.types import Asset
+from pysrc.util.enum_conversions import enum_to_string
+from pathlib import Path
+
+def asset_to_model_filename(asset: Asset) -> str:
+    return(f"{enum_to_string(asset)}_lasso.pkl")
```
Comment (echavemann) [inline src/pysrc/util/lasso_utils.py:6]: run formatter

Comment (Jiarui-Cao) [reply src/pysrc/util/lasso_utils.py]: yup

File: src/pysrc/util/lasso_utils.py [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+from pysrc.util.enum_conversions import enum_to_string
+from pathlib import Path
+
+
+def asset_to_model_filename(asset: Asset) -> str:
+    return f"{enum_to_string(asset)}_lasso.pkl"
+
+
+def search_model_in_directory(directory_path: Path, assets: list[Asset]) -> bool:
+    for asset in assets:
```
Comment (aidanvillasenor) [inline src/pysrc/util/lasso_utils.py:11]: not something to deal with now, but we're looping through assets a LOT in test cases. not sure if there's a way we can reduce how much we're duplicating code but maybe somethign we should think about

Comment (aidanvillasenor) [reply src/pysrc/util/lasso_utils.py:11]: reason i bring this up is bc i did the same thing lol

Comment (aidanvillasenor) [reply src/pysrc/util/lasso_utils.py:11]: wait nvm this isnt a test file disregard this lol

**Review by echavemann — APPROVED** [review]

**Review by aidanvillasenor — APPROVED** [review]

File: src/pysrc/util/lasso_utils.py (hunk 7 above)
Comment (Jiarui-Cao) [reply src/pysrc/util/lasso_utils.py:11]: Makes sense

**Pull request merged**